File: clases 25-11-2025/Tipos de clases js/clase normal con interfaz/Clases/Profesor.js
import IProfesor from '../interfaces/IProfesor.js';

export default class Profesor extends IProfesor {
    constructor(nombre, materia) {
        super();
        this.nombre = nombre;
        this.materia = materia;
    }

    mostrarProfesor() {
        console.log(`Profesor: ${this.nombre}, Materia: ${this.materia}`);
    }
}

File: clases 25-11-2025/Tipos de clases GO/clase_normal_con_interfaz/Profesor.go
package main

import "fmt"

type IProfesor interface {
	MostrarProfesor()
}

type Profesor struct {
	nombre  string
	materia string
}

func NewProfesor(nombre, materia string) IProfesor {
	return Profesor{nombre: nombre, materia: materia}
}

func (p Profesor) MostrarProfesor() {
	fmt.Printf("Profesor: %s, Materia: %s\n", p.nombre, p.materia)
}
